server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1144,19 +1180,17 @@ private Engine.IndexResult applyIndexOperation(
         SourceToParse sourceToParse,
         String id
     ) throws IOException {
-
-        // For Segment Replication enabled replica shards we can be skip parsing the documents as we directly copy segments from primary
-        // shard.
-        if (indexSettings.isSegRepEnabledOrRemoteNode() && routingEntry().primary() == false) {
+        // Only use segment replication path if sourceToParse is null
+        if (indexSettings.isSegRepEnabledOrRemoteNode() && routingEntry().primary() == false && sourceToParse == null) {
```

Comment:
I'm confused about these changes. This looks like a logical change unrelated to this feature?

Reply:
You're right, I need not make these for my changes. But many test cases were failing. As you mentioned, this is not a required feature. I can remove this. I added to ensure all the test cases are covered without any failure.